### PR TITLE
Spearhead: archive block pattern

### PR DESCRIPTION
**Pull request by MaggieCabrera**
#### Changes proposed in this Pull Request:

This PR is broken on the frontend because of this issue with Gutenberg (https://github.com/Automattic/themes/issues/2560) but I went ahead and finished the styles for the search block on the editor to at least have that. When the gutenberg issue is fixed, the frontend should look like the editor

* Search block
* Styles for the archive/category pages (the excerpt post styles go on a different PR: https://github.com/Automattic/themes/pull/2575 )

Screenshot 2020-10-09 at 19 39 28

* Added the category block + a title to the block pattern:

Screenshot 2020-10-13 at 12 24 07

* Cleaned up a bit the Search Block styles for other posible positions of the button with text/icon. This will probably need some design @kjellr :

Screenshot 2020-10-13 at 12 22 59

#### Related issue(s):

#2524

### CODE REVIEW

**Review by jffng — COMMENTED** [review]
The search block looks great in the editor, nice work!

I wonder if we should add some padding to the button when the icon style is not active? 

Screen Shot 2020-10-09 at 3 00 27 PM

**Comment by MaggieCabrera** [comment]
Waiting on [this PR](https://github.com/WordPress/gutenberg/pull/26052) to be merged for the search block fixes

File: spearhead/assets/sass/_archive.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.archive {
+
+    .page-header {
+        line-height: 56px;
```
Comment (scruffian) [inline spearhead/assets/sass/_archive.scss:4]: I wonder if this should be an `em` value so it adapts to the font size variable

**Comment by kjellr** [comment]
> Cleaned up a bit the Search Block styles for other posible positions of the button with text/icon. This will probably need some design @kjellr

Are these block options? I don't see them. 🤔

**Comment by MaggieCabrera** [comment]
> Are these block options? I don't see them. 🤔

Yes, these are all new to Gutenberg 9.1, hence why they are buggy :D

**Comment by kjellr** [comment]
Ah, got it. Ok well here's how I think these variations _should_ look based on the other mockups: 

search

That's all in Figma too if you want to take a look there.

**Comment by MaggieCabrera** [comment]
> Ah, got it. Ok well here's how I think these variations _should_ look based on the other mockups:

Wow, so fast! Thank you, I'll get to it!

File: spearhead/assets/sass/block-extends/_categories.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+.wp-block-categories {
+	.cat-item {
+		line-height: 2.2em;
+		margin: 0 0 5px 0;
+	}
+
+	a {
+		color: var(--entry-header--color);
+		font-size: var(--latest-posts--title-font-size);
+		font-weight: var(--heading--font-weight);
```
Comment (MaggieCabrera) [inline spearhead/assets/sass/block-extends/_categories.scss:17]: Looking at the figma these are `font-size: 20px` and `font-weight: 500`.

File: spearhead/inc/block-patterns.php [hunk 3]
```diff
@@ -35,6 +35,19 @@ function spearhead_register_block_patterns() {
...
 				)
 			);
+
+			register_block_pattern(
+				'spearhead/archive-page',
+				array(
+					'title'      => __( 'Archive page', 'spearhead' ),
+					'categories' => array( 'spearhead' ),
+					'content'    => '<!-- wp:search {"label":"Search","showLabel":false,"placeholder":"Search...","buttonText":"Search","buttonPosition":"button-inside","buttonUseIcon":true} /-->
+					<!-- wp:spacer {"height":50} --><div style="height:50px" aria-hidden="true" class="wp-block-spacer"></div><!-- /wp:spacer -->
```
Comment (MaggieCabrera) [inline spearhead/inc/block-patterns.php:45]: I think the spacer could be quite smaller according to the spacing changes suggested by @kjellr. Right now the space between the line of the input and the title is 120px, on the figma file it's 80px.

**Comment by MaggieCabrera** [comment]
We still need to address the styles for the [possible versions of the search block](https://github.com/Automattic/themes/pull/2564#issuecomment-707878367) but I think it would be fair to open a new issue/PR for them

File: spearhead/assets/sass/block-extends/_search.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+	// wrong class on gutenberg editor, we'll need to change it after that's fixed
+	&.wp-block-search__icon-button.wp-block-search__button-inside {
+		.wp-block-search__button {
+			padding: var(--button--padding-vertical) var(--button--padding-horizontal);
+		}
+	}
+
+	// wrong class on gutenberg editor, we'll need to change it after that's fixed
+	&.wp-block-search__text-button.wp-block-search__button-inside {
```
Comment (MaggieCabrera) [inline spearhead/assets/sass/block-extends/_search.scss:16]: Lines 9 and 16 need to swap after the gutenberg bug was solved

**Comment by scruffian** [comment]
I updated the layout based on the figma file:
Screenshot 2020-10-15 at 12 17 26

I decided to make the "titles" 18px as that's the base font size, and then use 22px for the links so that they are noticeably bigger than the headings. This also means we can reuse existing variables which makes the theme more consistent generally.

I think we should follow up with the search block styles in a different PR.

**Comment by MaggieCabrera** [comment]
Looks good to me!

**Comment by scruffian** [comment]
We'll follow up with the search block changes in https://github.com/Automattic/themes/issues/2606.

**Pull request merged**